### PR TITLE
Fixed a logger exception from assigning a null pointer to a string

### PR DESCRIPTION
**Pull request by mleotta**
Also cleaned up the code a little to avoid the need for ifdefs

### CODE REVIEW

File: vital/logger/kwiver_logger_manager.cxx [hunk 1]
```diff
@@ -101,34 +101,24 @@ ::kwiver_logger_manager()
   // we provide a method for creating these loggers.
 
   bool try_default(false);
-  std::string factory = std::getenv( PLUGIN_ENV_VAR );
```
Comment (mleotta) [inline vital/logger/kwiver_logger_manager.cxx]: @aaron-bray This line was causing segfaults when the environment variable is not set.  The results was a large number of failing unit tests.  We need to check that the return value is not null before assigning the `char*` to a `std::string`.

File: vital/logger/kwiver_logger_manager.cxx [hunk 2]
```diff
@@ -101,34 +101,24 @@ ::kwiver_logger_manager()
...
-#if defined(WIN32)
-    factory += ".dll";
-#elif defined(__APPLE__)
-    factory += ".so";
-#else
-    factory += ".so";
-#endif
+    factory_name = factory;
   }
+  factory_name += DL::LibExtension();
```
Comment (mleotta) [inline vital/logger/kwiver_logger_manager.cxx]: I also factored out all the `#ifdef`s by letting the dynamic loader tell me what the appropriate extension should be.  Let me know if this doesn't work for you.

**Comment by aaron-bray** [comment]
Yeah, That makes sense
My bad

Thanks for finding this

**Comment by aaron-bray** [comment]
That dynamic loader class is pretty cool! It handles a lot of the leg work in a generic way!
I think I may use it for other projects!

**Pull request merged**